From a83f3cb3360a8c1fd975ffb501b87bac41eb1c68 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:34 -0400
Subject: [fs] btrfs: disable snapshot aware defrag for now

Message-id: <1407179109-13714-148-git-send-email-sandeen@redhat.com>
Patchwork-id: 87098
O-Subject: [RHEL7.1 PATCH 147/422] Btrfs: disable snapshot aware defrag for now
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Josef Bacik <jbacik@fb.com>

commit 8101c8dbf6243ba517aab58d69bf1bc37d8b7b9c
Author: Josef Bacik <jbacik@fb.com>
Date:   Wed Jan 29 16:05:30 2014 -0500

    Btrfs: disable snapshot aware defrag for now

    It's just broken and it's taking a lot of effort to fix it, so for now just
    disable it so people can defrag in peace.  Thanks,

    Cc: stable@vger.kernel.org
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/inode.c b/fs/btrfs/inode.c
index b2ecca4..d07606e 100644
--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -2629,7 +2629,7 @@ static int btrfs_finish_ordered_io(struct btrfs_ordered_extent *ordered_extent)
    EXTENT_DEFRAG, 1, cached_state);
  if (ret) {
   u64 last_snapshot = btrfs_root_last_snapshot(&root->root_item);
-  if (last_snapshot >= BTRFS_I(inode)->generation)
+  if (0 && last_snapshot >= BTRFS_I(inode)->generation)
    /* the inode is shared */
    new = record_old_file_extents(inode, ordered_extent);
 
-- 
1.7.1